symfony/src/Controller/AccessManagementController.php

```diff
@@ -32,16 +47,102 @@ public function userGroups()
     /**
      * @Route("/user/group/{groupId}", name="access_manage_user_group")
      */
-    public function userGroup($groupId, GoogleGroupsService $googleGroupsService)
+    public function userGroup($groupId)
     {
         $group = $this->getUser()->getSiteFromId($groupId);
         if (empty($group)) {
             throw $this->createNotFoundException();
         }
-        $members = $googleGroupsService->getMembers($group->email);
+        $members = $this->googleGroupsService->getMembers($group->email);
         return $this->render('accessmanagement/group-members.html.twig', [
             'group' => $group,
             'members' => $members
         ]);
     }
+
+    /**
+     * @Route("/user/group/{groupId}/member", name="access_manage_user_group_member")
+     */
+    public function member($groupId, Request $request)
+    {
+        $group = $this->getUser()->getSiteFromId($groupId);
+        if (empty($group)) {
+            throw $this->createNotFoundException();
+        }
+        $groupMemberForm = $this->createForm(GroupMemberType::class);
+        $groupMemberForm->handleRequest($request);
+        if ($groupMemberForm->isSubmitted()) {
+            if ($groupMemberForm->isValid()) {
+                $email = $groupMemberForm->get('email')->getData() . self::MEMBER_DOMAIN;
+                $result = $this->googleGroupsService->addMember($group->email, $email);
+                if ($result['status'] === 'success') {
+                    $this->addFlash('success', $result['message']);
+                    $this->loggerService->log(Log::GROUP_MEMBER_ADD, [
+                        'member' => $email,
+                        'result' => 'success'
+                    ]);
+                    return $this->redirectToRoute('access_manage_user_group', ['groupId' => $groupId]);
+                }
+                $this->addFlash('error', $result['message']);
+                $this->loggerService->log(Log::GROUP_MEMBER_ADD, [
+                    'member' => $email,
+                    'result' => 'fail'
+                ]);
```

Comment:
Same as below. Can we return a generic error message to the user, and then add the return the result message to the log object?